packages/go/analysis/azure/post.go

```diff
@@ -993,26 +1027,63 @@ func ExecuteCommand(ctx context.Context, db graph.Database) (*analysis.AtomicPos
 	}
 }
 
-func resetPassword(ctx context.Context, db graph.Database, operation analysis.StatTrackedOperation[analysis.CreatePostRelationshipJob], roleAssignments RoleAssignments) error {
-	usersWithoutRoles := roleAssignments.UsersWithoutRoles()
+func resetPassword(ctx context.Context, db graph.Database, operation analysis.StatTrackedOperation[analysis.CreatePostRelationshipJob], tenant *graph.Node, roleAssignments RoleAssignments) error {
+	return operation.Operation.SubmitReader(func(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob) error {
+		if pwResetRoles, err := TenantRoles(tx, tenant, ResetPasswordRoleIDs()...); err != nil {
+			return err
+		} else {
+			for _, role := range pwResetRoles {
+				if roleTemplateIDProp := role.Properties.Get(azure.RoleTemplateID.String()); roleTemplateIDProp.IsNil() {
+					log.Errorf("unable to process azresetpassword for role node %d - missing property %s", role.ID, azure.RoleTemplateID)
```

Comment:
I feel like if we're going to use our pipeline pattern, we should really early return in each error branch (`return nil`), to prevent surprises if something happens after the pipeline in a refactor.